onnxruntime/core/framework/allocatormgr.h

```diff
@@ -40,24 +36,20 @@ struct AllocatorCreationInfo {
 // Valid values can be found in onnxruntime_c_api.h.
 AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info);
 
-// TODO: Only used for TRT and CUDA EP currently, need to add more identifiers to use it across all EPs
+// Used for sharing allocators across EPs. e.g. CUDA with TensorRT, CPU with XNNPACK
 class AllocatorManager {
   //
  public:
   AllocatorManager() = default;
   void InsertAllocator(AllocatorPtr allocator);
   void ReplaceAllocator(AllocatorPtr allocator);
-  //Get an allocator with specified device id and MemType. Return nullptr if it doesn't exist
-  AllocatorPtr GetAllocator(int id, OrtMemType mem_type) const;
+  // Get an allocator for the device. Return nullptr if it doesn't exist
+  AllocatorPtr GetAllocator(OrtMemType mem_type, OrtDevice device) const;
 
  private:
+  // key from OrtMemType+OrtDevice mapped to AllocatorPtr
+  using AllocatorMap = std::unordered_map<int32_t, AllocatorPtr>;
```

Comment:
So,  this class, even it is called "manager", doesn't actually own the allocator?